commit 85176a3fcd9748558cff72d4cdff5465b8732282
Author: Johannes Berg <johannes.berg@intel.com>
Date:   Wed Sep 21 23:01:46 2022 +0200

    wifi: mac80211: check link ID in auth/assoc continuation
    
    Ensure that the link ID matches in auth/assoc continuation,
    otherwise we need to reset all the data.
    
    Fixes: 81151ce462e5 ("wifi: mac80211: support MLO authentication/association with one link")
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>

diff --git a/net/mac80211/ieee80211_i.h b/net/mac80211/ieee80211_i.h
index 517a50abdb09..b704656027db 100644
--- a/net/mac80211/ieee80211_i.h
+++ b/net/mac80211/ieee80211_i.h
@@ -390,6 +390,7 @@ struct ieee80211_mgd_auth_data {
 	bool done, waiting;
 	bool peer_confirmed;
 	bool timeout_started;
+	int link_id;
 
 	u8 ap_addr[ETH_ALEN] __aligned(2);